core/src/main/java/com/yahoo/oak/Chunk.java

```diff
@@ -913,14 +1030,20 @@ private void findNewAnchor() {
             } else if (anchor == 1) { // cannot get below the first index
                 anchor = entrySet.getHeadNextIndex();
             } else {
-                if ((anchor - SKIP_ENTRIES_FOR_BIGGER_STACK) > 1) {
+                if ((anchor - skipEntriesForBiggerStack) > 1) {
                     // try to skip more then one backward step at a time
                     // if it shows better performance
-                    anchor -= SKIP_ENTRIES_FOR_BIGGER_STACK;
+                    anchor -= skipEntriesForBiggerStack;
```

Comment:
Performance: note that this might cause us to miss opportunities to avoid boundary checks.

Reply:
If I understand it correctly you are worried that we will check also the keys before midIdx's key. The only alternative that I see is to check for midIdx on every next step (as in ascending scan) and this I would like to avoid. We can discuss it during the code review meeting.